test/Common/TestEntityClasses.cs

```diff
@@ -52,5 +72,39 @@ public DateTime Post(string content)
                 return this.ChatEntries.ToList();
             }
         }
+
+        //-------------- An entity representing a counter object -----------------
+
+        [JsonObject(MemberSerialization = MemberSerialization.OptIn)]
+        public class Counter : ICounter
+        {
+            [JsonProperty("value")]
+            public int Value { get; set; }
+
+            public void Increment()
+            {
+                this.Value += 1;
+            }
+
+            public void Add(int value)
+            {
+                this.Value += value;
+            }
+
+            public async Task<int> Get()
+            {
+                return this.Value;
```

Comment:
Maybe do Task.FromResult(this.Value) here and remove the `async` from the method? Otherwise the compiler complains.